tests/integration/targets/aws_s3_bucket_info/tasks/main.yml

```diff
@@ -0,0 +1,92 @@
+---
+- name: Test community.aws.aws_s3_bucket_info
+  module_defaults:
+    group/aws:
+      aws_access_key: "{{ aws_access_key }}"
+      aws_secret_key: "{{ aws_secret_key }}"
+      security_token: "{{ security_token | default(omit) }}"
+      region: "{{ aws_region }}"
+  block:
+    - name: Create simple s3_buckets
+      s3_bucket:
+        name: "{{ item }}"
+        state: present
+        tags:
+          integration: "verified"
+          facts: "defined"
+      register: output
+      loop: "{{ testing_buckets }}"
+
+    - name: Get simple S3 bucket list
+      aws_s3_bucket_info:
+      register: bucket_list
+
+    - name: Assert result.changed == False and bucket list was retrieved
+      assert:
+        that:
+          - bucket_list.changed == False
+          - bucket_list.buckets
+
+    - name: Get complex S3 bucket list
+      aws_s3_bucket_info:
+        name_filter: "{{ name_pattern }}"
+        bucket_facts:
+          bucket_accelerate_configuration: true
+          bucket_acl: true
+          bucket_cors: true
+          bucket_encryption: true
+          bucket_lifecycle_configuration: true
+          bucket_location: true
+          bucket_logging: true
+          bucket_notification_configuration: true
+          bucket_ownership_controls: true
+          bucket_policy: true
+          bucket_policy_status: true
+          bucket_replication: true
+          bucket_request_payment: true
+          bucket_tagging: true
+          bucket_website: true
+          public_access_block: true
+        transform_location: true
+      register: bucket_list
+
+    - name: Assert that buckets list contains requested bucket facts
+      assert:
+        that:
+          - item.name is search(name_pattern)
+          - item.bucket_accelerate_configuration is defined
+          - item.bucket_acl is defined
+          - item.bucket_cors is defined
+          - item.bucket_encryption is defined
+          - item.bucket_lifecycle_configuration is defined
+          - item.bucket_location is defined
+          - item.bucket_logging is defined
+          - item.bucket_notification_configuration is defined
+          - item.bucket_ownership_controls is defined
+          - item.bucket_policy is defined
+          - item.bucket_policy_status is defined
+          - item.bucket_replication is defined
+          - item.bucket_request_payment is defined
+          - item.bucket_tagging is defined
+          - item.bucket_website is defined
+          - item.public_access_block is defined
+      loop: "{{ bucket_list.buckets }}"
+      loop_control:
+        label: "{{ item.name }}"
+
+    - name: Assert that retrieved bucket facts contains valid data
+      assert:
+        that:
```

Comment:
It would be good to see a little more by way of testing the actual values.  Location for example  (especially since you have a flag for changing how location's changed)

Reply:
https://github.com/zeten30/community.aws/commit/5e2231d4984e9fbed4a3c372a715819b8c560805